R/local-moran-bv-impl.R

```diff
@@ -12,16 +12,17 @@
 #'
 #' @inheritParams moran_bv_calc
 #' @keywords internal
-local_moran_bv_calc <- function(x, yj, wt) {
-  x * mapply(function(wij, yj) sum(wij * yj), wt, yj)
+#' x * st_lag(y, nb, wt)
+local_moran_bv_calc <- function(x, y, nb, wt) {
+  x * st_lag(y, nb, wt)
 }
 
 # local_moran_bv_calc(scale(x), find_xj(y, nb), wt)
 
 #' Local Bivariate Moran's I conditional permutation implementation
 #' @inheritParams moran_bv_perm_impl
 #' @keywords internal
-local_moran_bv_perm_impl <- function(x, y, listw) {
+local_moran_bv_perm_impl <- function(x, y, nb, listw) {
```

Comment:
We do not need to change the other function definitions. 
For context, the `nb` list and `wt` list objects are combined into another list object called `listw` which is used by `{spdep}`

Reply:
Inside this function on line 33 it calls `local_moran_bv_calc()` We need to update the function call so it will work with the new definition. Additionally, since we changed the definition to use the vector y, we do not need to create the object `p_yj`. For some more context, `p_yj` is a list object that contains the y values for the neighbors of x. 

`st_lag()` takes care of all that for us.

So! 

1. remove line 31 with `p_yj`
2. modify line 33 to use new function syntax e.g. `local_moran_bv_calc(x, y, nb, wt)`